src/context.rs

```diff
@@ -38,6 +42,7 @@ type Callback = dyn FnOnce(&mut Context);
 pub struct Context {
     plan_queue: Queue<Box<Callback>>,
     callback_queue: VecDeque<Box<Callback>>,
+    event_handlers: Option<HashMap<TypeId, Box<dyn Any>>>,
```

Comment:
Can you improve the comments about why we need an Option

Reply:
I replaced the use of Option here with destructuring.